### PR TITLE
Add training code

### PR DESCRIPTION
**Pull request by qyliuAI**

### CODE REVIEW

File: vision/wukong-huahua/ldm/models/autoencoder.py [hunk 1]
```diff
@@ -14,8 +14,10 @@
 # ============================================================================
 import mindspore as ms
 import mindspore.nn as nn
+import mindspore.ops as P
```
Comment (congw729) [inline vision/wukong-huahua/ldm/models/autoencoder.py:17]: Please make sure all the places using "ms.ops" are replaced by "P"

File: vision/wukong-huahua/ldm/models/diffusion/ddpm.py [hunk 2]
```diff
@@ -12,7 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ============================================================================
-import time
 import mindspore as ms
 import mindspore.nn as nn
 import mindspore.ops as ops
```
Comment (congw729) [inline vision/wukong-huahua/ldm/models/diffusion/ddpm.py:17]: Please make sure the consistent abbreviation of using "mindspore.ops": 
the file vision/wukong-huahua/ldm/models/autoencoder.py uses "P"; this one uses "ops".

File: vision/wukong-huahua/ldm/models/diffusion/dpm_solver/dpm_solver.py [hunk 3]
```diff
@@ -1,3 +1,17 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
 import math
 import mindspore as ms
 from mindspore import ops
```
Comment (congw729) [inline vision/wukong-huahua/ldm/models/diffusion/dpm_solver/dpm_solver.py:17]: Please make sure the consistent  **abbreviation** and **writing style** of using "mindspore.ops": 
1. the file vision/wukong-huahua/ldm/models/autoencoder.py uses "P"; this one uses "ops".
2. the file vision/wukong-huahua/ldm/models/diffusion/ddpm.py writes "import mindspore.ops as ops".

File: vision/wukong-huahua/ldm/modules/diffusionmodules/model.py [hunk 4]
```diff
@@ -17,17 +17,21 @@
 import mindspore.nn as nn
 import mindspore.ops as P
```
Comment (congw729) [inline vision/wukong-huahua/ldm/modules/diffusionmodules/model.py:18]: The same problem needs to be fixed here.

**Review by congw729 — COMMENTED** [review]
Please check the comments.

File: vision/wukong-huahua/ldm/models/autoencoder.py (hunk 1 above)
Comment (qyliuAI) [reply vision/wukong-huahua/ldm/models/autoencoder.py:17]: 经讨论，下次重构时一同整改

File: vision/wukong-huahua/ldm/models/diffusion/ddpm.py (hunk 2 above)
Comment (qyliuAI) [reply vision/wukong-huahua/ldm/models/diffusion/ddpm.py:17]: 经讨论，下次重构时一同整改

File: vision/wukong-huahua/ldm/models/diffusion/dpm_solver/dpm_solver.py (hunk 3 above)
Comment (qyliuAI) [reply vision/wukong-huahua/ldm/models/diffusion/dpm_solver/dpm_solver.py:17]: 经讨论，下次重构时一同整改

File: vision/wukong-huahua/ldm/modules/diffusionmodules/model.py (hunk 4 above)
Comment (qyliuAI) [reply vision/wukong-huahua/ldm/modules/diffusionmodules/model.py:18]: 经讨论，下次重构时一同整改

**Pull request merged**